proposals/user-scripts-execute-api.md

```diff
@@ -0,0 +1,150 @@
+In the User Script API proposal, executing a user script one time was [presented](https://github.com/w3c/webextensions/blob/main/proposals/user-scripts-api.md#execute-user-scripts-one-time) as a potential future enhancement.
+
+## API Overview
+
+### Objective
+
+Allow extensions to inject user scripts programmatically into web content given a target context.
+
+#### Use Cases
+
+The user wants to inject a user script on the page currently loaded in a given tab one time, but does not want to inject the script on that site automatically in the future. That is, the parameter that determines the injection is a tab ID and not based on origin.
+
+### Availability: Public
+
+This API will be public to all extensions (with the appropriate permission).
+
+### Consumers
+
+Any existing user script manager consumers of `chrome.tabs.executeScript()` that inject user scripts programmatically are intended to migrate to this new version in Manifest V3.
+
+## Implementation
+
+### API Schema
+
+#### Types
+
+```
+dictionary UserScriptInjection {
+  // Whether the injection should be triggered in the target as soon as
+  // possible. Note that this is not a guarantee that injection will occur
+  // prior to page load, as the page may have already loaded by the time the
+  // script reaches the target.
+  injectImmediately?: boolean,
+  // The script source to inject into the target.
+  js: ScriptSource,
+  // Details specifying the target into which to inject the script.
+  target: InjectionTarget,
+  // The JavaScript "world" to run the script in. The default is `USER_SCRIPT`.
+  world?: ExecutionWorld
+}
+
+dictionary InjectionTarget {
+  // Whether the script should inject into all frames within the tab. Defaults
+  // to false. This must not be true if `frameIds` is specified.
+  allFrames?: boolean,
+  // The IDs of specific documentIds to inject into. This must not be set if
+  // frameIds is set.
+  documentIds?: string[]
+  // The IDs of specific frames to inject into.
+  frameIds?: number[]
+  // The ID of the tab into which to inject.
+  tabId: number
+}
+
+dictionary InjectionResult {
+  // The document associated with the injection.
+  documentId: string,
+  // The frame associated with the injection.
+  frameId: number,
+  // The result of the script execution.
+  result?: any
+  // The error, if any. error and result are mutually exclusive.
+  error?: any
+}
+```
+
+#### Methods
+
+```
+// Injects a script into a target context. The script will be run at
+// `document_idle` unless the `injectImmediately` property is set. If the
+// script evaluates to a promise, the browser will wait
+// for the promise to settle and return the resulting value.
+Promise<InjectionResult[]> <browser>.userScripts.execute(
```

Comment:
Since we're working towards standardizing the `browser` namespace, let's just use it instead of the `<browser>` placeholder.

```suggestion
Promise<InjectionResult[]> browser.userScripts.execute(
```